utils/firestore.js

```diff
@@ -1960,21 +1953,15 @@ const retrieveNotificationSchemaByID = async (id) => {
 };
 
 const retrieveNotificationSchemaByCategory = async (category, getDrafts = false) => {
-  let result = [];
-  let query = db.collection("notificationSpecifications");
-  if (category !== "all") query = query.where("category", "==", category);
-  else query = query.orderBy("category");
-
-  // TODO: update query to get daft schemas directly, after all schema in Firestore are updated to have isDraft field
-  const snapshot = await query.orderBy("attempt").get();
-  if (snapshot.size === 0) return result;
-
-  for (const doc of snapshot.docs) {
-    const docData = doc.data();
-    if ((getDrafts && docData.isDraft) || (!getDrafts && !docData.isDraft)) result.push(docData);
+  let query = db.collection("notificationSpecifications").where("isDraft", "==", getDrafts);
+  if (category !== "all") {
+    query = query.where("category", "==", category);
+  } else {
+    query = query.orderBy("category");
   }
 
-  return result;
+  const snapshot = await query.orderBy("attempt").get();
```

Comment:
will this `orderBy` override the one in the previous `else` statement? or does it act as an additional `orderBy` similar to how a SQL query would work?

Reply:
I believe it works similarly to SQL's way. [Here](https://github.com/firebase/snippets-web/blob/e74e8fd9591f70007a44c58580ff0f1fcd1c7959/firestore/test.firestore.js#L942-L942) is an example used in its documentation.

Reply:
was a new index required? if so, we will want to remember when code goes to each environment

Reply:
Yes. Created dev and stage indexes already. Need your help creating a prod index.